main.js

```diff
@@ -10,42 +11,65 @@ import {
   getAssetSizes,
 } from './lib/helpers';
 
-let octokit;
-
 async function run() {
   try {
-    const myToken = getInput('repo-token', { required: true });
-    octokit = new GitHub(myToken);
+    const { myToken, cwd, usePrArtifacts } = getActionInputs();
+
+    const octokit = new GitHub(myToken);
+
     const pullRequest = await getPullRequest(context, octokit);
+    const fileDiffs = await diffAssets({ pullRequest, cwd, usePrArtifacts });
+
+    await commentOnPR({ octokit, pullRequest, fileDiffs })
+  } catch (error) {
+    setFailed(error.message);
+  }
+}
+
+async function getActionInputs() {
+  const workingDirectory = getInput('working-directory', { required: false });
+  const usePrArtifacts = getInput('use-pr-artifacts', { required: false });
+  const myToken = getInput('repo-token', { required: true });
 
-    const prAssets = await getAssetSizes();
+  const cwd = path.join(process.cwd(), workingDirectory);
```

Comment:
this new var gets passed around everywhere. the `process.cwd()` is the repo-root, but sometimes we want to change the cwd.